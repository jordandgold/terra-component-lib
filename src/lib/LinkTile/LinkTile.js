import React from "react";
import "./LinkTile.scss";
import PropTypes from "prop-types";
import Icon from "../Icon/Icon";

const LinkTile = ({ content }) => {
  const { title, text, link } = content;

  return (
    <article className="ter-link-tile">
      <div className="ter-link-tile__link-wrapper">
        <a className="ter-link-tile__title" href={link.url}>
          {title}
        </a>
        <Icon type="enclosed-caret-right-dark-24px" size="24px" />
      </div>
      <p className="ter-link-tile__body">{text}</p>
    </article>
  );
};

export default LinkTile;

LinkTile.propTypes = {
  content: PropTypes.shape({
    title: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    link: PropTypes.shape({
      url: PropTypes.string.isRequired
    }).isRequired
  })
};
