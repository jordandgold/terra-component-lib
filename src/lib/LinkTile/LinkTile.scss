@import "../global-styles/terra.scss";

.ter-link-tile {
  max-width: 500px;
  padding: 16px;

  &--link-wrapper {
    margin-bottom: 16px;
  }

  &--title {
    font-size: font-size("md");
    margin-right: 8px;
  }
}
