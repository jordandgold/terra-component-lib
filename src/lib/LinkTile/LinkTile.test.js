import React from "react";
import LinkTile from "./LinkTile";
import { shallow } from "enzyme";

describe("LinkTile", () => {
  let wrapper;

  beforeEach(() => {
    const mockContent = {
      title: "Design is good. Period.",
      text:
        "Forget the fat lady! You're obsessed with the fat lady! Drive us out of here!",
      link: {
        url: "#"
      }
    };
    wrapper = shallow(<LinkTile content={mockContent} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
