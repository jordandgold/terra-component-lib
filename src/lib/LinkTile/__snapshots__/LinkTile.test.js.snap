// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkTile should match the snapshot 1`] = `
<article
  className="ter-link-tile"
>
  <div
    className="ter-link-tile__link-wrapper"
  >
    <a
      className="ter-link-tile__title"
      href="#"
    >
      Design is good. Period.
    </a>
    <Icon
      size="24px"
      type="enclosed-caret-right-dark-24px"
    />
  </div>
  <p
    className="ter-link-tile__body"
  >
    Forget the fat lady! You're obsessed with the fat lady! Drive us out of here!
  </p>
</article>
`;
